admin-app/dictionary/views.py

```
@@ -0,0 +1,150 @@
+@require_GET
+def check_searchengine_status(request):
+    try:
+        client = get_client()
+        if not client.ping():
+            return JsonResponse({'status': 'unavailable', 'indexExists': False, 'docCount': 0}, status=503)
+
+        index_exists, doc_count = get_index_status(client, get_index_name())
+        return JsonResponse({'status': 'available', 'indexExists': index_exists, 'docCount': doc_count})
+    except (os_exc.ConnectionTimeout, os_exc.ConnectionError, TimeoutError):
+        return JsonResponse({'status': 'unavailable', 'indexExists': False, 'docCount': 0}, status=503)
+    except Exception as e:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)